core/controllers/classifier.py

```diff
@@ -123,3 +126,29 @@ def post(self):
         classifier_services.mark_training_job_complete(job_id)
 
         return self.render_json({})
+
+
+class NextJobHandler(base.BaseHandler):
+    """ This handler fetches next job in job queue and sends back job_id,
+    algorithm_id and training data to the VM.
+    """
+
+    def post(self):
+        """ Handels POST requests. """
+        signature = self.payload.get('signature')
+        vm_id = self.payload.get('vm_id')
+        if vm_id == feconf.DEFAULT_VM_ID and not feconf.DEV_MODE:
+            raise self.UnauthorizedUserException
+        if not verify_signature(None, vm_id, signature):
+            raise self.UnauthorizedUserException
+        next_job = classifier_services.fetch_next_job()
+        training_data = classifier_services.fetch_training_data(
+            next_job.exp_id, next_job.state_name)
+        classifier_services.update_training_job_training_data(
+            next_job.id, training_data)
+        classifier_services.mark_training_job_pending(next_job.id)
+        return self.render_json({
```

Comment:
If there are no new jobs then `next_job` will be None. In that case this line will generate exception. So render only when there is next job otherwise `self.render_json({})`. I have already taken care of empty response on Oppia-ml side.